### PR TITLE
 Functionality implemented for supporting XEP-0092

### PR DESCRIPTION
**Pull request by mnsuccess**
The functionality implemented for responding to the version of the software (XEP-0092)

### CODE REVIEW

File: src/main/java/nl/goodbytes/xmpp/xep0363/Component.java [hunk 1]
```diff
@@ -114,6 +118,29 @@ protected IQ handleIQGet( IQ iq ) throws Exception
...
         final Element request = iq.getChildElement();
         final Collection<String> namespaces = Arrays.asList( NAMESPACE, NAMESPACE_EXP );
+        // Implements the TYPE_IQ jabber:iq:version protocol (version info xep-0092). Allows
+        // XMPP entities to query each other's application versions.  The server
+        // will respond with its current version info.
+        if ("query".equals(request.getQName().getName()) && "jabber:iq:version".equals( request.getNamespaceURI())) {
+            try {
+                Element answerElement = DocumentHelper.createElement(QName.get("query", "jabber:iq:version"));
+                answerElement.addElement("name").setText("Http File Upload Component");
+                answerElement.addElement("version").setText("1.2.0");
```
Comment (guusdk) [inline src/main/java/nl/goodbytes/xmpp/xep0363/Component.java]: For bonus points, try to make this use the version number that is defined in the Maven pom.xml file. If we don't do that, then I will likely forget to update this code, every time we do a new release!

**Comment by mnsuccess** [comment]
> Why does this PR introduce a new submodule (named 'bin')?

File: pom.xml [hunk 2]
```diff
@@ -48,6 +48,19 @@
     </developer>
   </developers>
 
+  <contributors>
+        <contributor>
+            <name>Manasse Ngudia</name>
+            <roles>
+                <role>developper contributor</role>
```
Comment (guusdk) [inline pom.xml]: Change this to "developer" (one letter 'p') please. Don't include word "contributor"

File: src/main/java/nl/goodbytes/xmpp/xep0363/Component.java [hunk 3]
```diff
@@ -114,6 +120,33 @@ protected IQ handleIQGet( IQ iq ) throws Exception
...
+        // will respond with its current version info.
+        if ("query".equals(request.getQName().getName()) && "jabber:iq:version".equals( request.getNamespaceURI())) {
+            try {
+                Element answerElement = DocumentHelper.createElement(QName.get("query", "jabber:iq:version"));
+                MavenXpp3Reader reader = new MavenXpp3Reader();
+                Model model = reader.read(new FileReader("pom.xml"));
+                answerElement.addElement("name").setText(model.getName());
+                answerElement.addElement("description").setText(model.getDescription());
+                answerElement.addElement("version").setText(model.getVersion());
+                answerElement.addElement("Organization").setText(model.getOrganization().getName());
```
Comment (guusdk) [inline src/main/java/nl/goodbytes/xmpp/xep0363/Component.java]: "Organization" is not a field defined in the XEP, so we'd better leave it off. Also, if the POM does not define an organization, I expect that `model.getOrganization().getName()` to result in a NullPointerException. It is good to explicitly check for these things.

**Comment by guusdk** [comment]
I don't know why, but that submodule does not want to go away.

Maybe it's easier if you create a completely new PR, based on a fresh copy of the code?

**Comment by mnsuccess** [comment]
> I don't know why, but that submodule does not want to go away.
> 
> Maybe it's easier if you create a completely new PR, based on a fresh copy of the code?

okay i'm going to create a new PR

**Pull request closed without merging**